api/user/author/model.py

```diff
@@ -1,27 +1,34 @@
-from dataclasses import dataclass
-from datetime import datetime
+from dataclasses import asdict, dataclass
 
 from flask_login import UserMixin
 from flask_sqlalchemy import SQLAlchemy
 from sqlalchemy import event
 
 from api import db
+from api.user.relations import author_likes_comments, author_likes_posts
+from api.utils import generate_object_ID
+
+
+def _constructURL(context):
+    host = context.get_current_parameters()["host"]
+    id = context.get_current_parameters()["id"]
+    host = host + "/" if not host.endswith("/") else host
+    return host + "authors/" + id
 
 
 @dataclass
-class Author(UserMixin, db.Model):
-    id: int = db.Column(db.Integer, primary_key=True)
-    url: str = db.Column("url", db.Text, nullable=True)
+class Author(db.Model):
```

Comment:
```suggestion
class Author(UserMixin, db.Model):
```

Got removed. Is necessary for login to work I think